diff --git a/roaring/src/bitmap/store/interval_store.rs b/roaring/src/bitmap/store/interval_store.rs
index 86b2022..4c944a0 100644
--- a/roaring/src/bitmap/store/interval_store.rs
+++ b/roaring/src/bitmap/store/interval_store.rs
@@ -855,13 +855,6 @@ impl<I: SliceIterator<Interval>> Iterator for RunIter<I> {
     }
 
     fn nth(&mut self, n: usize) -> Option<Self::Item> {
-        if n > usize::from(u16::MAX) {
-            // Consume the whole iterator
-            self.intervals.nth(self.intervals.as_slice().len());
-            self.forward_offset = 0;
-            self.backward_offset = 0;
-            return None;
-        }
         if let Some(skip) = n.checked_sub(1) {
             let mut to_skip = skip as u64;
             loop {
