diff --git a/roaring/src/bitmap/store/interval_store.rs b/roaring/src/bitmap/store/interval_store.rs
index 86b2022..0b3d15e 100644
--- a/roaring/src/bitmap/store/interval_store.rs
+++ b/roaring/src/bitmap/store/interval_store.rs
@@ -734,10 +734,6 @@ impl<I: SliceIterator<Interval>> RunIter<I> {
             }
             Err(index) => {
                 if index == self.intervals.as_slice().len() {
-                    // Consume the whole iterator
-                    self.intervals.nth(index);
-                    self.forward_offset = 0;
-                    self.backward_offset = 0;
                     return;
                 }
                 if let Some(value) = index.checked_sub(1) {
