diff --git a/roaring/src/bitmap/store/interval_store.rs b/roaring/src/bitmap/store/interval_store.rs
index 86b2022..4f107e2 100644
--- a/roaring/src/bitmap/store/interval_store.rs
+++ b/roaring/src/bitmap/store/interval_store.rs
@@ -656,7 +656,6 @@ impl<I: SliceIterator<Interval>> RunIter<I> {
         if let Some(value) = self.forward_offset.checked_add(1) {
             self.forward_offset = value;
         } else {
-            self.intervals.next();
             self.forward_offset = 0;
             return;
         }
